orc8r/cloud/go/services/bootstrapper/bootstrapper/main.go

```diff
@@ -34,26 +39,60 @@ var (
 func main() {
 	srv, err := service.NewOrchestratorService(orc8r.ModuleName, bootstrapper.ServiceName)
 	if err != nil {
-		glog.Fatalf("Error creating service: %+v", err)
+		glog.Fatalf("error creating service: %+v", err)
+	}
+
+	bs := createBootstrapperServicer()
+	crs, rs := createRegistrationServicers()
+
+	protos.RegisterBootstrapperServer(srv.GrpcServer, bs)
+	protos.RegisterCloudRegistrationServer(srv.GrpcServer, crs)
+	protos.RegisterRegistrationServer(srv.GrpcServer, rs)
+
+	err = srv.Run()
+	if err != nil {
+		glog.Fatalf("error running service: %+v", err)
 	}
+}
 
+func createBootstrapperServicer() (*servicers.BootstrapperServer) {
 	key, err := key.ReadKey(*keyFilepath)
 	if err != nil {
-		glog.Fatalf("Error reading bootstrapper private key: %+v", err)
+		glog.Fatalf("error reading bootstrapper private key: %+v", err)
 	}
 	rsaPrivateKey, ok := key.(*rsa.PrivateKey)
 	if !ok {
-		glog.Fatalf("Error coercing bootstrapper private key to RSA private key; actual type: %T", key)
+		glog.Fatalf("error coercing bootstrapper private key to RSA private key; actual type: %T", key)
 	}
 
 	servicer, err := servicers.NewBootstrapperServer(rsaPrivateKey)
 	if err != nil {
-		glog.Fatalf("Error creating bootstrapper servicer: %+v", err)
+		glog.Fatalf("error creating bootstrapper servicer: %+v", err)
 	}
-	protos.RegisterBootstrapperServer(srv.GrpcServer, servicer)
+	return servicer
+}
 
-	err = srv.Run()
+func createRegistrationServicers() (protos.CloudRegistrationServer, protos.RegistrationServer) {
+	db, err := sqorc.Open(storage2.GetSQLDriver(), storage2.GetDatabaseSource())
 	if err != nil {
-		glog.Fatalf("Error running service: %+v", err)
+		glog.Fatalf("failed to connect to database: %s", err)
```

Comment:
Always template err with `%+v` to give full info eg stack trace

Reply:
there's a ton of `%s` for err everywhere?

Reply:
Correct, there are 🙂